src/Storages/MergeTree/ExportPartitionManifestUpdatingTask.cpp

```diff
@@ -0,0 +1,328 @@
+#include <Storages/MergeTree/ExportPartitionManifestUpdatingTask.h>
+#include <Storages/StorageReplicatedMergeTree.h>
+#include <Storages/ExportReplicatedMergeTreePartitionTaskEntry.h>
+#include "Storages/MergeTree/ExportPartitionUtils.h"
+#include "Common/logger_useful.h"
+#include <Common/ZooKeeper/Types.h>
+#include <Common/ZooKeeper/ZooKeeper.h>
+#include <Interpreters/DatabaseCatalog.h>
+
+namespace DB
+{
+
+struct CleanupLockRAII
```

Comment:
Apparently, there is a construct made just for that: https://github.com/Altinity/ClickHouse/blob/antalya-25.8/src/Common/ZooKeeper/ZooKeeper.h#L730